.github/workflows/android-build.yml

```diff
@@ -13,101 +13,89 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    # Checkout the repository
-    - name: Checkout repository
-      uses: actions/checkout@v4
+      # Step 1: Checkout the repository
+      - name: Checkout repository
+        uses: actions/checkout@v4
 
-    # Set up JDK 17 for the Android build
-    - name: Set up JDK 17
-      uses: actions/setup-java@v4
-      with:
-        java-version: '17'
-        distribution: 'zulu'
+      # Step 2: Set up JDK 17
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'zulu'
 
-    # Cache Gradle dependencies
-    - name: Cache Gradle packages
-      uses: actions/cache@v4
-      with:
-        path: |
-          ~/.gradle/caches
-          ~/.gradle/wrapper
-        key: gradle-${{ runner.os }}-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
-        restore-keys: |
-          gradle-${{ runner.os }}-
+      # Step 3: Cache Gradle dependencies
+      - name: Cache Gradle packages
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: gradle-${{ runner.os }}-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            gradle-${{ runner.os }}-
 
-    # Set up Android SDK
-    - name: Set up Android SDK
-      uses: android-actions/setup-android@v2
-      with:
-        api-level: 34
-        build-tools: "34.0.0"
-        ndk: "25.1.8937393"
+      # Step 4: Set up Android SDK
+      - name: Set up Android SDK
+        uses: android-actions/setup-android@v2
+        with:
+          api-level: 34
+          build-tools: "34.0.0"
+          ndk: "25.1.8937393"
 
-    # Decode and create google-services.json from the secret
-    - name: Decode google-services.json
-      env:
-        GOOGLE_SERVICES_JSON: ${{ secrets.GOOGLE_SERVICES_JSON }}
-        MAPS_API_KEY: ${{ secrets.MAPS_API_KEY }}
-      run: |
-        echo $GOOGLE_SERVICES_JSON | base64 --decode > ./app/google-services.json
-        echo "$MAPS_API_KEY" | base64 --decode > ./local.properties      
+      # Step 5: Decode and place google-services.json
+      - name: Decode google-services.json
+        env:
+          GOOGLE_SERVICES_JSON: ${{ secrets.GOOGLE_SERVICES_JSON }}
+        run: |
+          echo "$GOOGLE_SERVICES_JSON" | base64 --decode > ./app/google-services.json
 
-      # Debug: Check if google-services.json is correctly decoded
-    - name: Debug google-services.json
-      run: cat ./app/google-services.json
-      
-    # Grant execute permissions to gradlew
-    - name: Grant execute permission for gradlew
-      run: chmod +x ./gradlew
+      # Step 6: Place MAPS API Key in local.properties
```

Comment:
Good practice to separate google-service.json and local.properties in two different steps